### PR TITLE
fix: client shutdown when auth is disabled and connection fails

### PR DESCRIPTION
**Pull request by jdogmcsteezy**
* Includes logging improvements

### CODE REVIEW

File: connection_provider.go [hunk 1]
```diff
@@ -396,6 +396,8 @@ func (cp *connectionProvider) connectToSeeds(ctx context.Context) error {
 				if err != nil {
 					logger.WarnContext(ctx, "failed to connect to seed", slog.Any("error", err))
 
+					authErr = err
```
Comment (reugn) [inline connection_provider.go:399]: A non-synchronized write of `authErr` (also in the previous if). There is no guarantee that the reader goroutine will observe the change.

File: connection_provider_test.go [hunk 2]
```diff
@@ -98,7 +98,11 @@ func TestNewConnectionProvider_FailConnectToSeedConns(t *testing.T) {
 	}
 
 	assert.Nil(t, cp)
-	assert.Equal(t, "failed to connect to seeds: context deadline exceeded", err.Error())
+	assert.Equal(
+		t,
+		"failed to connect to seeds: server error: DeadlineExceeded, msg=context deadline exceeded",
```
Comment (reugn) [inline connection_provider_test.go:103]: The previous error message is more idiomatic.

Comment (jdogmcsteezy) [reply connection_provider_test.go]: you are right

File: connection_provider.go (hunk 1 above)
Comment (jdogmcsteezy) [reply connection_provider.go]: synchronized with sync.Once

Comment (reugn) [reply connection_provider.go]: Actually `wg.Wait()` and `close(seedConns)` ensure synchronization.

**Review by reugn — APPROVED** [review]

File: connection_provider.go (hunk 1 above)
Comment (jdogmcsteezy) [reply connection_provider.go]: I thought there would still be a race condition on write? I don't care which err is saved but I still thought there could be issue with multiple concurrent writes

**Pull request merged**